elf_reader.go

```diff
@@ -393,19 +395,51 @@ func (ec *elfCode) loadFunctions(section *elfSection) (map[string]asm.Instructio
 			// When splitting sections into subprograms, the targets of these calls
 			// are no longer in scope, so they must be resolved here.
 			if ins.IsFunctionReference() && ins.Constant != -1 {
-				tgt := jumpTarget(offset, ins)
+				tgt := jumpTarget(secOffset, ins)
 				sym := section.symbols[tgt].Name
 				if sym == "" {
-					return nil, fmt.Errorf("offset %d: no jump target found at offset %d", offset, tgt)
+					return nil, fmt.Errorf("offset %d: no jump target found at offset %d", secOffset, tgt)
 				}
 
-				ins.Reference = sym
+				ins.SetReference(sym)
 				ins.Constant = -1
 			}
 		}
 
+		// Only rebuild lineInfos at the start of a function
```

Comment:
What about a subprog's FuncInfo? Could we also pull that in here?

Reply:
Yea, this would also be the location to do the same for FuncInfo. That is something we will have to add before we can marshall metadata back into wire formatted LineInfo and FuncInfo. 

FuncInfo is nothing more than a name and BTF type, however I skipped this because dealing with references to BTF might have some implications. Especially on how to marshall a reference back to type IDs in the final blob, and if we would need additional info. So I decided to not implement the whole funcInfo part yet until we have a better plan for the marshalling part. I have no use case for it as of now since we already have symbols which are basically the same just without the BTF type.